kedro/io/kedro_data_catalog.py

```diff
@@ -112,6 +114,50 @@ def __eq__(self, other) -> bool:  # type: ignore[no-untyped-def]
             other.config_resolver.list_patterns(),
         )
 
+    def keys(self, regex_search: str | None = None) -> List[str]:  # noqa: UP006
+        return self._filter_keys(regex_search)
+
+    def values(self, regex_search: str | None = None) -> List[AbstractDataset]:  # noqa: UP006
+        return [self._datasets[key] for key in self._filter_keys(regex_search)]
```

Comment:
Just a nitpick, because the syntax is neat,  but to be honest I can't think about any use case that would need to handle values (i.e. datasets objects) without the associated key (i.e. the dataset name). 

Say you have ``catalog.values(regex_search="^country_")`` :  it is very likely that I want to know the exact data (hence its name) I am loading / reassigning / whatever modification I do, and not just iterate over all datasets  matching the criteria. 

A much better filter search would be something specific to the dataset itself (e.g. its type, some metadata it has...), but I definitely understand it is out of scope here. I am not sure adding other filtering is better than just letting people do something like this : 

```python
[ds_name for ds_name in catalog if isinstance(catalog[ds_name], PandasDataset)] 
```

Reply:
That totally makes sense.

We have a ticket on improving search options - https://github.com/kedro-org/kedro/issues/3917, so there's a high chance it will be modified in the future anyway. We also discussed whether we should provide any options for filtering since now it will be easy for users to apply it on top of the keys returned as in your example.

Currently, we kept it as part of the previous API - `list()` method. I added it for `values()` to align with `keys()` and `items()`. However, I see it leading to confusion, so I will remove it from the interface for `values()`.